[action] Continue git_commit action execution if there is nothing to commit

mollyIV opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
[`git_commit`](https://docs.fastlane.tools/actions/git_commit/) action fails if there's nothing to commit. 

<img width="589" alt="git_commit_nothing_to_commit" src="https://user-images.githubusercontent.com/10795657/55280686-39ccd300-5329-11e9-96f3-cf301792ae7c.png">

<!-- If it fixes an open issue, please link to the issue here. -->
closes https://github.com/fastlane/fastlane/issues/14437

### Description
<!-- Describe your changes in detail -->

Added a new parameter `allow_nothing_to_commit`. 

<!-- Please describe in detail how you tested your changes. -->

Tested the changes by adding a new unit test and doing manual testing. 

<img width="455" alt="Screenshot 2019-03-30 at 19 27 40" src="https://user-images.githubusercontent.com/10795657/55280688-481aef00-5329-11e9-9754-1ce21026b713.png">

<img width="511" alt="Screenshot 2019-03-30 at 19 27 03" src="https://user-images.githubusercontent.com/10795657/55280692-51a45700-5329-11e9-8dee-71f017679269.png">

mollyIV committed:
[action] Continue git_commit action execution if there is nothing to commit

janpio commented on fastlane/lib/fastlane/actions/git_commit.rb:
```diff
@@ -10,8 +10,23 @@ def self.run(params)
+          result = Actions.sh(
+            command,
+            log: FastlaneCore::Globals.verbose?,
+            error_callback: lambda { |_|
```
Is "nothing to commit" the only reason this can error?

joshdholtz replied:
This catching of error callback might be a bit to aggressive. It _will_ allow nothing to get committed but it will also eat all/any other errors as well 🤔 Might need to look into another solution 🙃

mollyIV committed:
Code review changes

mollyIV commented:
👋 @janpio and @joshdholtz 

Thanks for your questions / suggestions. I updated the implementation - now we're checking if there's something to commit via `git status --porcelain`. If the git index is clean, there's no point of committing changes, so we just early return. 

Added one more unit tests 💪

Let me know what you think 🙇

joshdholtz commented on fastlane/lib/fastlane/actions/git_commit.rb:
```diff
@@ -10,8 +10,17 @@ def self.run(params)
-        result = Actions.sh("git commit -m #{params[:message].shellescape} #{paths} #{skip_git_hooks}".strip)
-        UI.success("Successfully committed \"#{params[:path]}\" 💾.")
+        command = "git commit -m #{params[:message].shellescape} #{paths} #{skip_git_hooks}".strip
+        nothing_to_commit = false
```
I don't think we need this. The if statement is early returning out of the method don't need this 😊

joshdholtz commented on fastlane/lib/fastlane/actions/git_commit.rb:
```diff
@@ -10,8 +10,17 @@ def self.run(params)
+        end
+
+        result = Actions.sh(command)
+        UI.success("Successfully committed \"#{params[:path]}\" 💾.") unless nothing_to_commit
```
Can remove the `unless nothing_to_commit` since the clean repo is early returning 💣

joshdholtz commented on fastlane/lib/fastlane/actions/git_commit.rb:
```diff
@@ -10,8 +10,17 @@ def self.run(params)
 
-        result = Actions.sh("git commit -m #{params[:message].shellescape} #{paths} #{skip_git_hooks}".strip)
-        UI.success("Successfully committed \"#{params[:path]}\" 💾.")
+        command = "git commit -m #{params[:message].shellescape} #{paths} #{skip_git_hooks}".strip
```
We could probably move this line after the `if params[:allow_nothing_to_commit]` block since this line isn't needed until after that 😊

joshdholtz reviewed (changes requested):
This implementation looks much better! Just a few minor code nitpicks 😊

mollyIV committed:
Clean up

mollyIV commented:
👋 @joshdholtz 

PR is ready for another round, thanks for your comments! 💪

joshdholtz reviewed (approved):
This is 🔥 Thanks again for another contribution! ❤️
